godot-core/src/builtin/collections/packed_array.rs

```diff
@@ -414,29 +437,50 @@ macro_rules! impl_packed_array {
         #[doc = concat!("Creates a `", stringify!($PackedArray), "` from the given slice.")]
         impl From<&[$Element]> for $PackedArray {
             fn from(slice: &[$Element]) -> Self {
-                let mut array = Self::new();
-                let len = slice.len();
-                if len == 0 {
-                    return array;
+                if slice.is_empty() {
+                    return Self::new();
                 }
-                array.resize(len);
-                let ptr = array.ptr_mut(0);
-                for (i, element) in slice.iter().enumerate() {
-                    // SAFETY: The array contains exactly `len` elements, stored contiguously in memory.
-                    unsafe {
-                        // `GString` does not implement `Copy` so we have to call `.clone()`
-                        // here.
-                        *ptr.offset(to_isize(i)) = element.clone();
-                    }
+                let mut array = Self::default_with_size(slice.len());
+                // SAFETY: The array contains exactly `len` elements, stored contiguously in memory.
+                let dst = unsafe { std::slice::from_raw_parts_mut(array.ptr_mut(0), slice.len()) };
+                dst.clone_from_slice(slice);
+                array
```

Comment:
Also here, some vertical space would help:
```suggestion

                let mut array = Self::default_with_size(slice.len());
                
                // SAFETY: The array contains exactly `len` elements, stored contiguously in memory.
                let dst = unsafe { std::slice::from_raw_parts_mut(array.ptr_mut(0), slice.len()) };
                dst.clone_from_slice(slice);
                array
```

Reply:
Done